database_wrapper.py

```diff
@@ -616,12 +618,14 @@ def update_entity(self, entity_type, data_dict: dict, filter_fields: list) -> bo
 
         # If we're not filtering for anything, we shouldn't be calling update_entity
         if len(filter_fields) == 0:
-            raise RuntimeError("update_entity for {} requires filter_fields list to filter by.".format(entity_type))
+            raise RuntimeError(
+                "update_entity for {} requires filter_fields list to filter by.".format(
+                    entity_type
+                )
+            )
```

Comment:
I didn't notice that running `black` did this. Personally, I'm not a huge fan of this style for python - any other thoughts/opinions?

Reply:
Honestly, as long as I don't have to think about formatting then I'm happy. 
Feel free to [experiment with `black`'s configuration](https://black.readthedocs.io/en/stable/pyproject_toml.html)

By the way, now that we're thinking about formatting 😅 

Here's a possibly more readable alternative:
```python
        if len(filter_fields) == 0:
            msg = "update_entity for {} requires filter_fields list to filter by."
            raise RuntimeError(msg.format(entity_type))
```

Here's another that uses [f-strings](https://realpython.com/python-f-strings/)
```python
        if len(filter_fields) == 0:
            msg = f"update_entity for {entity_type} requires filter_fields list to filter by."
            raise RuntimeError(msg)
```

I believe `black` will leave both of these alone. [Here is evidence](https://black.now.sh/?version=stable&state=_Td6WFoAAATm1rRGAgAhARYAAAB0L-Wj4AG0ALtdAD2IimZxl1N_Wlws4TBexXdmg613D2cRmcZ76uq_UPwvHBlZFGoLImlyrakg2VDS7cptnsaFpx55YPB8RgTxe8ieJjpGuyLLAtvWv_qw6F-iJmENCh_VS4IPUWWxReaFe3cJ1uCxNqkWPCP80F2oTZBjIe8jFhQ3JjX5WguivFNJx7WvC1V6fq9DjeeOTcCeaJzbrQ3BcFRZTvLuzDXQ1PEC_PvmpdpRFqQwJCi6NIBo1unqvxz_KL4zdSkAAL6J8QnmwlAEAAHXAbUDAAAL0eTZscRn-wIAAAAABFla)

Reply:
## [You could even make your own error](https://stackoverflow.com/questions/6180185/custom-python-exceptions-with-error-codes-and-error-messages)

```
class UpdateEntityError(Exception):
    def __init__(self, entity_type):
        self.entity_type = entity_type
    def __str__(self):
        msg = "update_entity for {} requires filter_fields list to filter by."
        return repr(msg.format(self.entity_type))
```

## How is this useful?
**Notice how I now only have to pass in the `entity_type` when raising.**
Formatting is taken care of for me. 

```
>>> try:
...     raise UpdateEntityError(100)
... except UpdateEntityError as e:
...     print("what is e?", e)
...     print("what is e.args?", e.args)
...     print("is e.args[0] the entity_type?", e.args[0], "... seems legit...")
...
what is e? 'update_entity for 100 requires filter_fields list to filter by.'
what is e.args? (100,)
is e.args[0] the entity_type? 100 ... seems legit...
```

@Jason-Ku

Reply:
Thanks for all the helpful info! I'll look into black's configuration and see if I can't slip my way around some of those strange formats

Reply:
Thoughts about [autopep8](https://github.com/hhatto/autopep8)? If we're just trying to fit within PEP8 guidelines I think this is the way to go. `Black` does some nice things but I'm iffy on its treatment of vertical whitespace, it adds a lot of artificial length to our files. I've also taken a look at Google's code style formatter (their Java one is REALLY nice! Can't vouch for the python one), which seems pretty good but also makes things a bit longer. Heres a link to that, it's called [yapf](https://github.com/google/yapf).

Reply:
Whoops, I never really read the first line of the docs. Sorry for sending you on that ghost hunt @Jason-Ku 

<img width="400" alt="image" src="https://user-images.githubusercontent.com/11131250/76670816-dd6dbe00-654f-11ea-9b55-7da460d26c57.png">

**In retrospect...** 
**Pro:** No config means no arguments over style. 
  * The robot justs takes over the code 😂

**Con:** it might be annoying to not have config options, 
  * but you can always insert a `# fmt : off` comment

Reply:
This [post about `yapf`](https://news.ycombinator.com/item?id=17155048) worries me.

<img width="416" alt="image" src="https://user-images.githubusercontent.com/11131250/76670956-c54a6e80-6550-11ea-909a-da77b219d6f1.png">

Reply:
I just dug around a little bit and it seems that YAPF has actually changed formatting between different versions, which makes absolutely no sense to me. 

So our choices are limited to autopep8 and black. I personally prefer just auto-formatting to fit pep8 guidelines with `autopep8`, because I think the way `black` handles multi-line arguments is pretty gross. If there's more support for `black` on our team, I'd be happy to concede though!